test/go.mod

```diff
@@ -22,17 +22,17 @@ require (
 	github.com/vmware/govmomi v0.37.2
 	golang.org/x/crypto v0.23.0
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.29.4
-	k8s.io/apimachinery v0.29.4
-	k8s.io/client-go v0.29.4
-	k8s.io/component-base v0.29.4
+	k8s.io/api v0.29.5
+	k8s.io/apimachinery v0.29.5
+	k8s.io/client-go v0.29.5
+	k8s.io/component-base v0.29.5
 	k8s.io/klog/v2 v2.110.1
 	k8s.io/utils v0.0.0-20231127182322-b307cd553661
 	sigs.k8s.io/cluster-api v0.0.0-00010101000000-000000000000
 	sigs.k8s.io/cluster-api-provider-vsphere v0.0.0-00010101000000-000000000000
 	sigs.k8s.io/cluster-api/test v0.0.0-00010101000000-000000000000
 	sigs.k8s.io/controller-runtime v0.17.5
-	sigs.k8s.io/kind v0.22.0
+	sigs.k8s.io/kind v0.23.0
```

Comment:
I guess this one is okay in CAPV. We don't really depend on the kind version. But we can condier adding this to the ignore list in the dependabot config as well

Reply:
@chrischdi WDYT?

Reply:
Ack, we can also remove the direct dependency by copying over this single function:

https://github.com/kubernetes-sigs/kind/blob/main/pkg/errors/concurrent.go#L42 + using `kerrors` in there instead.

I could add this to https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/3010

Reply:
Let's do that, sounds good!

Reply:
(We'll rebase this PR after yours merged, then)